lib/barbeque/slack_notifier.rb

```diff
@@ -68,6 +72,18 @@ def job_retry_link(job_retry)
       def job_retry_url(job_retry)
         Barbeque::Engine.routes.url_helpers.job_execution_job_retry_url(job_retry.job_execution, job_retry, host: barbeque_host)
       end
+
+      def should_notify_failure?(job_execution_with_slack_notification)
+        unless job_execution_with_slack_notification.slack_notification.notify_failure_only_if_retry_limit_reached
+          return true
+        end
+
+        unless job_execution_with_slack_notification.job_definition.retry_config
+          return true
+        end
+
+        job_execution_with_slack_notification.job_definition.retry_config.retry_limit == job_execution_with_slack_notification.job_retries.count
```

Comment:
(IMO) should be compared with `<=` for fail-safe

Reply:
Agreed. https://github.com/cookpad/barbeque/pull/77/commits/82eff9d10466d527b98c6fd1b5c92dea2bacb1f8